concepts/Common/Integration Guides/70 Manage Azure Blob Storage with DevExtreme Components/40 FileUploader.md

```diff
@@ -0,0 +1,55 @@
+
+The `FileUploader` component can upload blobs to Azure Storage with the multi-part upload technique.
+
+1. **Preparation**. Generate an SAS (Shared Access Signature) token for the Azure Blob Storage account or container. Azure requires this token to grant you data upload authorization.
+    Add a server-side method that generates the necessary token:
+
+        object UploadBlob(string blobName) {
+            if (blobName.Contains("/"))
+                return CreateErrorResult("Invalid blob name.");
+
+            string prefix = Guid.NewGuid().ToString("N");
+            string fullBlobName = $"{prefix}_{blobName}";
+            var blob = Container.GetBlockBlobClient(fullBlobName);
+
+            if (blob.Exists() && blob.GetProperties().Value.ContentLength > MaxBlobSize) {
+                return CreateErrorResult();
+            }
+            if (blob.CanGenerateSasUri) {
+                var sasUri = blob.GenerateSasUri(BlobSasPermissions.Write, DateTimeOffset.UtcNow.AddHours(1));
+                return CreateSuccessResult(sasUri.AbsoluteUri);
+            } else {
+                        return CreateErrorResult("BlobClient cannot generate SasUri");
+            }
+        }
+
+2. **Chunk upload**. Upload chunks one by one. Azure assigns a unique block ID for each chunk. We receive this ID inside the `ETag` header.
+3. **Completion**. To complete the upload, send all the `ETag` headers you collected during step 2 to the server. This action stitches the chunks together.
+
+    The following client-side function uploads chunks to Azure, saves the block ID, and reports block ID data back to Azure alongside the last chunk:
```

Comment:
Is "**alongside** the last chunk" correct? Or is the block ID contained _inside_ this last chunk of data?

Reply:
alongside is correct — it's a separate property :)